stdlib/runtime/equal.gr

```diff
@@ -186,12 +187,12 @@ let rec heapEqualHelp = (heapTag, xptr, yptr) => {
     }
   }
 }, equalHelp = (x, y) => {
-  if ((x & Tags._GRAIN_GENERIC_TAG_MASK) != 0n) {
+  if (((x & Tags._GRAIN_GENERIC_TAG_MASK) != 0n) && ((y & Tags._GRAIN_GENERIC_TAG_MASK) != 0n)) {
     // Short circuit for non-pointer values
     x == y
   } else if (isNumber(x)) {
     // Numbers have special equality rules, e.g. NaN != NaN
-    numberEqual(x, y)
+    isNumber(y) && numberEqual(x, y)
```

Comment:
This check shouldn't be necessary. What's up with it?

Reply:
Forgot that this function wasn't exported. Fixed.